cosmos-gui/test/node/mysql_driver_test.js

```diff
@@ -0,0 +1,112 @@
+
+// Testing variables
+var idm_username = 'frb@tid.es';
+var username = 'frb1';
+var password = '12345';
+var hdfsQuota = 5;
+
+// Mock the mysql connection
+var connectionMock = {
+    connect: function(callback) {
+        return callback(null);
+    },
+    query: function(query, values, callback) {
+        query = query || '';
+        values = values || [];
+        callback = callback || function(error, result) {};
+
+        if (query.indexOf('INSERT INTO cosmos_user (idm_username, username, password, hdfs_quota)') > -1) {
+            return callback(null, []);
+        } else if (query.indexOf('UPDATE cosmos_user SET password') > -1) {
+            return callback(null, []);
+        } else if (query.indexOf('SELECT * from cosmos_user WHERE idm_username') > -1) {
+            return callback(null, [username]);
+        } else if (query.indexOf('SELECT * from cosmos_user WHERE username') > -1) {
+            return callback(null, [username]);
+        } else {
+            return callback('error', null);
+        } // if else if
+    },
+    end: function() {}
+};
+mysqlDriver.__set__('connection', connectionMock);
+
+// Tests suite
+describe('[mysqlDriver.connect] create a MySQL connection', function() {
+    it('should return null error', function () {
+        mysqlDriver.connect(function(error) {
+            assert.equal(null, error);
+        });
+    });
+});
+
+describe('[mysqlDriver.addUser] add a new user', function() {
+    it('should return null error and an empty result set', function() {
+        mysqlDriver.addUser(idm_username, username, password, hdfsQuota, function(error, result) {
+            assert.equal(null, error);
+            assert.equal(0, result.length);
+        });
+    });
+});
+
+describe('[mysqlDriver.addPassword] add a new password', function() {
+    it('should return null error and an empty result set', function() {
+        mysqlDriver.addPassword(idm_username, password, function(error, result) {
+            assert.equal(null, error);
+            assert.equal(0, result.length);
+        });
+    });
+});
+
+describe('[mysqlDriver.getUser] get a user by the idm user', function() {
+    it('should return null error and a result set containing ' + username, function() {
+        mysqlDriver.addPassword(idm_username, password, function(error, result) {
+            assert.equal(null, error);
+            assert.equal(username, result[0]);
+        });
+    });
+});
+
+describe('[mysqlDriver.getUserByCosmosUser] get a user by the cosmos user', function() {
+    it('should return null error and an result set containing ' + username, function() {
+        mysqlDriver.addPassword(idm_username, password, function(error, result) {
+            assert.equal(null, error);
+            assert.equal(username, result[0]);
+        });
+    });
+});
+
+describe('[mysqlDriver.close] close a connection', function() {
+    it('should finish', function() {
+        mysqlDriver.close(connectionMock);
+    });
+});
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cosmos/commit/76f9443cc2c3e45f5d3740772e50ba7266ef6fbb